user_threshold/models/ir_config_parameter.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import os
+
+from odoo import _, api, models
+from odoo.exceptions import AccessError
+
+MAX_DB_USER_PARAM = 'user.threshold.database'
+HIDE_THRESHOLD = str(os.environ.get('USER_THRESHOLD_HIDE', '')) == '1'
+
+
+class IrConfigParameter(models.Model):
+    _inherit = 'ir.config_parameter'
+
+    def _can_manipulate(self):
+        """
+        Check to see if the user is a member of the correct group
+         Returns:
+             True when the user is a member of the threshold manager group
+        """
+        return self.env.user.has_group(
+            'user_threshold.group_threshold_manager'
+        )
+
+    @api.multi
+    def unlink(self):
+        """
+        Override to disallow deletion of the user threshold parameter
+        when the user does not have the right access
+        """
+        for rec in self.filtered(lambda r: r.key == MAX_DB_USER_PARAM):
+            if not self._can_manipulate():
+                raise AccessError(_(
```

Comment:
This line and L47 are showing as untested because the user I'm using for testing is not a member of the required group and therefore cannot read `r.key` on L33 and L46 respectively, causing an AccessError before we ever hit this one. If anyone has any ideas on circumventing this, I'm all ears :)